iron-list.html

```diff
@@ -1033,14 +1033,24 @@
       if (!el || el._templateInstance.__key__ !== key) {
         return;
       }
-      
       if (dot >= 0) {
         path = this.as + '.' + path.substring(dot+1);
         el._templateInstance.notifyPath(path, value, true);
       } else {
+        // Update selection if needed
+        var currentItem = el._templateInstance[this.as];

```

Comment:
Keep the selection in sync  e.g.   if `items[0] = 'a';` and `selection[0] = 'a';` if the user does `set('item.0', 'b')`  then  `selection[0]` should be `b`